ext/vscode/src/commands/env.ts

```diff
@@ -9,42 +9,134 @@ import { quickPickWorkspaceFolder } from '../utils/quickPickWorkspaceFolder';
+    const selectedEnvironment = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliEnvironment>() : undefined;
+    const selectedFile = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliEnvironments>().context.configurationFile : selectedItem;
     let folder: vscode.WorkspaceFolder | undefined = (selectedFile ? vscode.workspace.getWorkspaceFolder(selectedFile) : undefined);
     if (!folder) {
         folder = await quickPickWorkspaceFolder(context, localize('azure-dev.commands.util.needWorkspaceFolder', "To run '{0}' command you must first open a folder or workspace in VS Code", 'env select'));
     }
     const cwd = folder.uri.fsPath;
 
-    let envData: EnvironmentInfo[] = [];
-    let errorMsg: string | undefined = undefined;
-    try {
-        envData = await getEnvironments(context, cwd);
-    } catch(err) {
-        errorMsg = parseError(err).message;
-        // Treated the same as no environments case
+    let name = selectedEnvironment?.name;
+
+    if (!name) {       
+        let envData: EnvironmentInfo[] = [];
+        try {
+            envData = await getEnvironments(context, cwd);
+        } catch(err) {
+            // Treated the same as no environments case
+        }
+        
+        if (envData.length === 0) {
+            void vscode.window.showInformationMessage(localize('azure-dev.commands.cli.env-delete.no-environments', 'There are no environments to delete.'));
+            return;
+        }
+        
+        const envChoices  = envData.map(d => ({ label: d.Name, data: d,} as IAzureQuickPickItem<EnvironmentInfo>));
+        const selectedEnv = await context.ui.showQuickPick(envChoices, {
+            canPickMany: false,
+            title: localize('azure-dev.commands.cli.env-delete.choose-environment', 'Which environment should be deleted?')
+        });
+
+        name = selectedEnv.data.Name;
     }
 
-    if (envData.length === 0) {
-        await promptCreateNewEnvironment(localize('azure-dev.commands.cli.env-select.no-environments', 'There are no environments to select. Would you like to create one?'), errorMsg);
-        return; // promptCreateNewEnvironment() will call newEnvironment() asynchronously if necessary
+    const deleteOption: vscode.MessageItem = { title: localize('azure-dev.commands.cli.env-delete.delete', 'Delete') };
+
+    const result = await vscode.window.showWarningMessage(
+        localize('', 'Are you sure you want to delete the {0} environment?', name),
+        { modal: true },
+        deleteOption);
+
+    if (result === deleteOption) {
+        const environmentDirectory = vscode.Uri.joinPath(folder.uri, '.azure', name);
+        
+        await vscode.workspace.fs.delete(environmentDirectory, { recursive: true, useTrash: false });
+        
+        // TODO: Update default environment, if necessary.
+        
+        // TODO: Use Azure Dev CLI to delete environment.
```

Comment:
I agree--the logic to update default environment and other corner cases during deletion should be handled by the CLI.

@ellismg do we have a work item to add `env delete` command?

Reply:
We have this one: https://github.com/Azure/azure-dev/issues/937

Reply:
I created this to track doing it in VSCode, and will add a comment in the code for it: https://github.com/Azure/azure-dev/issues/1554